component/otelcol/auth/sigv4/sigv4_test.go

```diff
@@ -0,0 +1,194 @@
+package sigv4_test
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/grafana/agent/component/otelcol/auth"
+	"github.com/grafana/agent/component/otelcol/auth/sigv4"
+	"github.com/grafana/agent/pkg/flow/componenttest"
+	"github.com/grafana/agent/pkg/river"
+	"github.com/grafana/agent/pkg/util"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/config/configauth"
+	"gotest.tools/assert"
```

Comment:
Please use `github.com/stretchr/testify/assert` for consistency with the assertion libraries we use